lib/rules/prefer-query-by-disappearance.ts

```diff
@@ -0,0 +1,184 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+
+import { createTestingLibraryRule } from '../create-testing-library-rule';
+import {
+  getPropertyIdentifierNode,
+  isArrowFunctionExpression,
+  isCallExpression,
+  isMemberExpression,
+  isFunctionExpression,
+  isExpressionStatement,
+  isReturnStatement,
+  isBlockStatement,
+} from '../node-utils';
+
+export const RULE_NAME = 'prefer-query-wait-disappearance';
+export type MessageIds = 'preferQueryByDisappearance';
+
+export default createTestingLibraryRule<[], MessageIds>({
```

Comment:
I think we should extract `Options` type, so it's consistent with all other rules
```suggestion
type Options = [];

export default createTestingLibraryRule<Options, MessageIds>({
```